### PR TITLE
Cartesian to Keplerian

### PR DESCRIPTION
**Pull request by eggrobin**

### CODE REVIEW

File: physics/kepler_orbit.hpp [hunk 1]
```diff
@@ -32,6 +32,10 @@ class KeplerOrbit {
               Body const& secondary,
               KeplerianElements<Frame> const& elements_at_epoch,
               Instant const& epoch);
+  KeplerOrbit(MassiveBody const& primary,
+              Body const& secondary,
+              RelativeDegreesOfFreedom<Frame> const& state_vectors,
```
Comment (pleroy) [inline physics/kepler_orbit.hpp]: `state_vectors` is not a very nice name.  I know, that's how it's called, but that doesn't make it better.  How about `primary_to_secondary`?

File: physics/kepler_orbit_body.hpp [hunk 2]
```diff
@@ -45,6 +55,73 @@ KeplerOrbit<Frame>::KeplerOrbit(
...
+  Vector<double, Frame> const x({1, 0, 0});
+  Vector<double, Frame> const z({0, 0, 1});
+  Bivector<double, Frame> const x_wedge_y({0, 0, 1});
+
+  Bivector<SpecificAngularMomentum, Frame> const h = Wedge(r, v) * Radian;
+  // The eccentricity vector has magnitude equal to the eccentricity, and points
+  // towards the periapsis.  This is a vector (the direction of the periapsis
+  // does not depend on the coordinate system).
+  Vector<double, Frame> const eccentricity_vector =
+      v * h / Radian / μ - Normalize(r);
```
Comment (pleroy) [inline physics/kepler_orbit_body.hpp]: Unit (Radian) at the end?

File: physics/kepler_orbit_body.hpp [hunk 3]
```diff
@@ -45,6 +55,73 @@ KeplerOrbit<Frame>::KeplerOrbit(
...
+  auto const& periapsis = eccentricity_vector;
+  Vector<SpecificAngularMomentum, Frame> const ascending_node = z * h;
+
+  // Maps [-π, π] to [0, 2π].
+  auto const positive_angle = [](Angle const& α) -> Angle {
+    return α > 0 * Radian ? α : α + 2 * π* Radian;
```
Comment (pleroy) [inline physics/kepler_orbit_body.hpp]: Space before star.

File: physics/kepler_orbit.hpp (hunk 1 above)
Comment (eggrobin) [inline physics/kepler_orbit.hpp]: I'm not fond of `primary_to_secondary`. If we find something nicer than the standard terminology we'll rename things.

File: physics/kepler_orbit_body.hpp (hunk 2 above)
Comment (eggrobin) [inline physics/kepler_orbit_body.hpp]: Done (we should really investigate incorporating that into * and Wedge).

File: physics/kepler_orbit_body.hpp (hunk 3 above)
Comment (eggrobin) [inline physics/kepler_orbit_body.hpp]: Done.

**Comment by pleroy** [comment]
Still lgtm

**Pull request merged**